backend/routes/historical.py

```
@@ -0,0 +1,710 @@
+
+        # Add examples to dataset
+        generator = HistoricalDatasetGenerator()
+        updated_dataset = generator.add_examples_to_dataset(
+            dataset, profile, analysis, request.count
+        )
+
+        if not updated_dataset:
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Failed to expand dataset",
+            )
```

Ruff (0.14.3)

Abstract `raise` to an inner function

(TRY301)